samples/bot-all-cards/csharp/BotAllCards/Dialogs/MainDialog.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using BotAllCards.Cards;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.Dialogs.Choices;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class MainDialog : ComponentDialog
+    {
+        protected string ConnectionName { get; }
+
+        public MainDialog(IConfiguration configuration)
+             : base(configuration["ConnectionName"])
+        {
+            ConnectionName = configuration["ConnectionName"];
+
+            // Define the main dialog and its related components.
+            AddDialog(new ChoicePrompt(nameof(ChoicePrompt)));
+            AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[]
+            {
+                ShowAllCardAsync,
+                SelectCardAsync,
+            }));
+
+            // The initial child Dialog to run.
+            InitialDialogId = nameof(WaterfallDialog);
+        }
+
+        // 1. Prompts the user if the user is not in the middle of a dialog.
+        // 2. Re-prompts the user when an invalid input is received.
+        private async Task<DialogTurnResult> ShowAllCardAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+
+            // Create the PromptOptions which contain the prompt and re-prompt messages.
+            // PromptOptions also contains the list of choices available to the user.
+            var options = new PromptOptions()
+            {
+                Prompt = MessageFactory.Text("What card would you like to see? You can click or type the card name"),
+                RetryPrompt = MessageFactory.Text("That was not a valid choice, please select a card or number from 1 to 9."),
+                Choices = lodingAllCards(),
+            };
+
+            // Prompt the user with the configured PromptOptions.
+            return await stepContext.PromptAsync(nameof(ChoicePrompt), options, cancellationToken);
+        }
+
+        // Send a Rich Card response to the user based on their choice.
+        // This method is only called when a valid prompt response is parsed from the user's response to the ChoicePrompt.
```

Comment:
Not the correct way of adding method summary 
use ///

Reply:
Fixed